class.jetpack.php

```diff
@@ -4736,6 +4739,26 @@ function authenticate_jetpack( $user, $username, $password ) {
 		return new WP_User( $token_details['user_id'] );
 	}
 
+	// Authenticates requests from Jetpack server to WP API endpoints.
+	// Uses the existing XMLRPC oAuth implementation.
+	function wp_json_authenticate( $error ) {
```

Comment:
As written, this is probably breaking other plugins that do API authentication (it prevents them from returning an error if their authentication credentials are invalid).  To fix it, restructure as follows (the first bit is the most important):

```php
if ( $error ) {
    return $error;
}

if ( ! isset( $_GET['token'] ) ) {
    return null;
}

// ...

return new WP_Error( 'rest_forbidden', 'Sorry. You are not allowed to do that', 403 );
```